src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/JQueryValueProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// An <see cref="IValueProvider"/> for jQuery formatted form data.
+    /// </summary>
+    public abstract class JQueryValueProvider : BindingSourceValueProvider, IEnumerableValueProvider
+    {
+        private readonly IDictionary<string, StringValues> _values;
+        private PrefixContainer _prefixContainer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JQueryFormValueProvider"/> class.
+        /// </summary>
+        /// <param name="bindingSource">The <see cref="BindingSource"/> of the data.</param>
+        /// <param name="values">The values.</param>
+        /// <param name="culture">The culture to return with ValueProviderResult instances.</param>
+        public JQueryValueProvider(
+            BindingSource bindingSource,
+            IDictionary<string, StringValues> values,
+            CultureInfo culture)
+            : base(bindingSource)
+        {
+            if (bindingSource == null)
+            {
+                throw new ArgumentNullException(nameof(bindingSource));
+            }
+
+            if (values == null)
+            {
+                throw new ArgumentNullException(nameof(values));
+            }
+
+            _values = values;
+            Culture = culture;
+        }
+
+        // Internal for testing
+        internal CultureInfo Culture { get; }
```

Comment:
This could just as well be public. It's passed in through a public API so it's not an implementation detail.

Reply:
Please add tests covering this property. One of the few differences between the two jQuery factories is the `Culture` chosen for the value providers -- `CultureInfo.Invariant` for query strings and `CurrentCulture` for forms.